packages/blocks/src/api/raw-handling/paste-handler.js

```diff
@@ -241,7 +236,29 @@ export function pasteHandler( { HTML = '', plainText = '', mode = 'AUTO', tagNam
 		// Allows us to ask for this information when we get a report.
 		console.log( 'Processed HTML piece:\n\n', piece );
 
-		return htmlToBlocks( { html: piece, rawTransforms } );
+		const pieceBlocks = htmlToBlocks( { html: piece, rawTransforms } );
+
+		if ( ! canUserUseUnfilteredHTML ) {
+			// Should run before `figureContentReducer`.
+			filters.unshift( iframeRemover );
+
+			return compact( map( pieceBlocks, ( pieceBlock ) => {
+				if ( pieceBlock.name !== 'core/html' ) {
+					return pieceBlock;
+				}
+
+				pieceBlock.attributes.content = deepFilterHTML( pieceBlock.attributes.content, filters, blockContentSchema );
```

Comment:
Do we need to run _all_ filters again?

Reply:
Not by the look of it. Stripped it down in 984d93225.